itest/lnd_channel_funding_fund_max_test.go

```diff
@@ -0,0 +1,277 @@
+package itest
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/lightningnetwork/lnd"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"github.com/lightningnetwork/lnd/lntest"
+	"github.com/lightningnetwork/lnd/lntest/node"
+	"github.com/lightningnetwork/lnd/lnwallet/chainfee"
+	"github.com/lightningnetwork/lnd/lnwire"
+	"github.com/stretchr/testify/require"
+)
+
+type chanFundMaxTestCase struct {
+	// name is the name of the target test case.
+	name string
+
+	// initialWalletBalance is the amount in Alice's wallet.
+	initialWalletBalance btcutil.Amount
+
+	// pushAmt is the amount to be pushed to Bob.
+	pushAmt btcutil.Amount
+
+	// feeRate is an optional fee in satoshi/bytes used when opening a
+	// channel.
+	feeRate btcutil.Amount
+
+	// expectedBalanceAlice is Alice's expected balance in her channel.
+	expectedBalanceAlice btcutil.Amount
+
+	// chanOpenShouldFail denotes if we expect the channel opening to fail.
+	chanOpenShouldFail bool
+
+	// expectedErrStr contains the expected error in case chanOpenShouldFail
+	// is set to true.
+	expectedErrStr string
+
+	// private denotes if the channel opening is announced to the network or
+	// not.
+	private bool
+}
+
+// testChannelFundMax checks various channel funding scenarios where the user
+// instructed the wallet to use all remaining funds.
+func testChannelFundMax(ht *lntest.HarnessTest) {
+	// Create two new nodes that open a channel between each other for these
+	// tests.
+	alice := ht.NewNode("Alice", nil)
+	defer ht.Shutdown(alice)
+
+	bob := ht.NewNode("Bob", nil)
+	defer ht.Shutdown(bob)
```

Comment:
can we instead use the standby Alice and Bob nodes that the Harness comes with?

Reply:
I think the standby nodes are funded by default with an initial wallet balance which is not ideal behavior for our tests here.

Reply:
gotcha 👍